### PR TITLE
New Azure specs ruleset package

### LINKED ISSUE #654: Add new ruleset package for spec repo
```[tasklist]
### Tasks
- [x] Implement new package
- [x] Merge and release new package
- [x] Update TypeSpec validation to enforce new ruleset names
- [x] Bulk migrate all specs to new ruleset
```

### PR DESCRIPTION
**Pull request by timotheeguerin**
implement #654 
Package designed to provide rulesets for azure specs.

There is a test that makes sure for a given ruleset that all the rules from dependent linter are explicitly enabled or disabled. This will ensure that we don't forget to register rules there.

### CODE REVIEW

File: packages/typespec-azure-specs-rulesets/package.json [hunk 1]
```diff
@@ -0,0 +1,65 @@
+{
+  "name": "@azure-tools/typespec-azure-specs-rulesets",
```
Comment (timotheeguerin) [inline packages/typespec-azure-specs-rulesets/package.json:2]: What do people think of the name

Comment (markcowl) [reply packages/typespec-azure-specs-rulesets/package.json:2]: Why not typespec-azure-rulesets?

I am not strong on any recommendation here, both are plain in their meaning. Just seems like all rulesets apply to specs.

File: packages/typespec-azure-specs-rulesets/package.json [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  "version": "0.42.0",
+  "author": "Microsoft Corporation",
+  "description": "TypeSpec ruleset for azure specs",
```
Comment (markcowl) [inline packages/typespec-azure-specs-rulesets/package.json:5]:
```suggestion
  "description": "TypeSpec ruleset for Azure specs",
```

**Review by markcowl — APPROVED** [review]

File: packages/typespec-azure-specs-rulesets/package.json (hunk 1 above)
Comment (timotheeguerin) [reply packages/typespec-azure-specs-rulesets/package.json]: yeah I like it, renamed

**Pull request merged**